### ISSUE TITLE
v0.53.3: environment yaml files: "cannot unmarshal !!str `<no value>` into bool" typing issues?

### ISSUE THREAD

**Issue by bitsofinfo** [roboll/helmfile#553]
helmfile version v0.53.3

I have an environment YAML like this

x.yaml
```
kubeContext: "mycontext"
wait: false
```

My helmfile is like this
```
environments:
  x:
    values:
      - x.yaml

helmDefaults:
  tillerNamespace: kube-system 
  kubeContext: {{ .Environment.Name }} 
  verify: false
  wait: {{ .Environment.Values.wait }} 
  timeout: 600
  recreatePods: true
  force: true
....
```

When I run `helmfile -e x -f app.yaml apply` I get 

```
could not deduce `environment:` block, configuring only .Environment.Name. error: failed to read app.yaml: reading document at index 1: yaml: unmarshal errors:
line XX: cannot unmarshal !!str `<no value>` into bool
error during app.yaml parsing: template: stringTemplate:11:30: executing "stringTemplate" at <.Environment.Values.kubeContext>: map
 has no entry for key "kubeContext"
```

**Comment by bitsofinfo** [comment]
I get a similar type of error if targeting `timeout` with an integer declared in an env yaml file. Looks like there is no support for anything outside of strings?

**Comment by mumoshu** [comment]
@bitsofinfo Hmmm... I'm pretty confused. Would you mind running it with debug logs like `helmfile --log-level=debug -e x -f app.yaml apply` and share the logs?

**Comment by bitsofinfo** [comment]
env.x.yaml
```
kubeContext: x
wait: false
```


env.y.yaml
```
kubeContext: y
wait: true
```
helmfiletest.yaml
```
environments:
  x:
    values:
      - env.x.yaml
  y:
    values:
      - env.y.yaml

helmDefaults:
  tillerNamespace: kube-system 
  kubeContext: {{ .Environment.Values.kubeContext }} 
  verify: false
  wait: {{ .Environment.Values.wait }}
  timeout: 600
  recreatePods: true
  force: true

releases:
```


```
helmfile -f helmfiletest.yaml -e x apply
```

```
could not deduce `environment:` block, configuring only .Environment.Name. error: failed to read helmfiletest.yaml: reading document at index 1: yaml: unma
rshal errors:
  line 13: cannot unmarshal !!str `<no value>` into bool
error during helmfiletest.yaml parsing: template: stringTemplate:11:30: executing "stringTemplate" at <.Environment.Values.kubeContext>: map has no entry f
or key "kubeContext"
```

**Comment by mumoshu** [comment]
@bitsofinfo Seems like we've caught by a golang text/template gotcha! Thanks for reporting. This is being fixed via #645
